src/cloudsc_fortran_pyiface/CMakeLists.txt

```diff
@@ -0,0 +1,132 @@
+    DESCRIPTION "Build the new Fortran version CLOUDSC driven by Python" DEFAULT ON
+    CONDITION Serialbox_FOUND OR HDF5_FOUND
+)
+
+if( HAVE_CLOUDSC_FORTRAN_PYIFACE )
+   message (STATUS "Setting up module directory")
+   ecbuild_enable_fortran(MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../module)
+   message (STATUS  ${CMAKE_CURRENT_BINARY_DIR}/../../module)
+
+    set (DWARF_CLOUDSC_LIB  dwarf-cloudsc-lib)
+    ecbuild_add_library( TARGET ${DWARF_CLOUDSC_LIB}
+        SOURCES
+            ./fortransrc/cloudsc_driver_mod.F90
+            ./fortransrc/cloudsc.F90
+        DEFINITIONS ${CLOUDSC_DEFINITIONS}
+    )
+    set (PYTHON_MODN cloudsc)
+
+     set(FORTRAN_PYTHON_COMMON_SOURCES
+            yomphyder
+            yoecldp
+            yoethf
+            yomcst
+            yoephli
+        )
+
+     set(FORTRAN_PYTHON_DWARF_SOURCES
+            cloudsc_driver_mod
+        )
+
+     set (F90WRAP_COMMON_SOURCES ${FORTRAN_PYTHON_COMMON_SOURCES})
+     set (F90WRAP_DWARF_SOURCES  ${FORTRAN_PYTHON_DWARF_SOURCES})
+     set (F2PY_COMMON_SOURCES ${FORTRAN_PYTHON_COMMON_SOURCES})
+     set (F2PY_DWARF_SOURCES  ${FORTRAN_PYTHON_DWARF_SOURCES})
+     list(TRANSFORM F90WRAP_COMMON_SOURCES PREPEND ${CMAKE_CURRENT_SOURCE_DIR}/../common/module/)
+     list(TRANSFORM F90WRAP_COMMON_SOURCES APPEND .F90 )
+     list(TRANSFORM F90WRAP_DWARF_SOURCES PREPEND ${CMAKE_CURRENT_SOURCE_DIR}/fortransrc/)
+     list(TRANSFORM F90WRAP_DWARF_SOURCES APPEND .F90 )
+     list(TRANSFORM F2PY_COMMON_SOURCES PREPEND f90wrap_)
+     list(TRANSFORM F2PY_COMMON_SOURCES APPEND .f90 )
+     list(TRANSFORM F2PY_DWARF_SOURCES PREPEND f90wrap_)
+     list(TRANSFORM F2PY_DWARF_SOURCES APPEND .f90 )
+
+  set(PYTHON_SOURCES
+            ./pythonsrc/dwarfdriver.py
+            ./pythonsrc/cloudsc_data.py
+        )
+
+   add_custom_target( pythonexecs ALL SOURCES ${PYTHON_SOURCES}
+         COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/pythonsrc  ${CMAKE_CURRENT_BINARY_DIR}/pythonsrc
+         COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_BINARY_DIR}/pythonsrc  ../../bin/pythonexec
+         COMMAND sh -c "rm -f ../../bin/pythonexec/*.f90"
+         COMMENT "Copy Python driver to Fortran kernel to executable directory"
+        )
+
+    target_link_libraries( ${DWARF_CLOUDSC_LIB} PRIVATE cloudsc-common-lib HDF5::HDF5 )
+    # Define the binary build target for this variant
+    ecbuild_add_executable( TARGET dwarf-cloudsc-fortran-pyiref
+        SOURCES
+            ./fortransrc/dwarf_cloudsc.F90
+        DEFINITIONS ${CLOUDSC_DEFINITIONS}
+    )
+
+    target_link_libraries( dwarf-cloudsc-fortran-pyiref PRIVATE cloudsc-common-lib ${DWARF_CLOUDSC_LIB})
+
+    target_include_directories(dwarf-cloudsc-fortran-pyiref PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+    target_include_directories(dwarf-cloudsc-fortran-pyiref PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/../common)
+
+    if( HAVE_OMP AND TARGET OpenMP::OpenMP_Fortran )
```

Comment:
Aren't things like HDF5 / OpenMP / MPI / OpenACC linkage inherited from libcommon these days? I don't think this is necessary, or am I missing something here?